onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -367,8 +367,18 @@ namespace {
 template <typename Loader>
 OrtStatus* CreateSessionImpl(_In_ const OrtEnv* env, _In_ const OrtSessionOptions* options,
                              Loader loader, _Outptr_ OrtSession** out) {
-  auto sess = std::make_unique<::onnxruntime::InferenceSession>(
-      options == nullptr ? onnxruntime::SessionOptions() : options->value, env->loggingManager);
+  std::unique_ptr<::onnxruntime::InferenceSession> sess;
+  if (options == nullptr || options->value.session_thread_pool_size <= 0) {
```

Comment:
I would prefer to be rather explicit about the usage of the env thread pool. Something like a session option that says use_env_threadpool. I don't think this condition really qualifies for the usage of the env thread pool.

Reply:
Even without this PR, they are already sharing threads.  If it worked before, why it doesn't work anymore? Why should I disable it?

Reply:
Because now we're being explicit about the existence of a env threadpool.

Reply:
But that increases the complexity.  I don't think it's necessary.